content/en/blog/2022/instrument-apache-httpd-server.md

```diff
@@ -0,0 +1,212 @@
+---
+title: Learn how to instrument Apache Http Server with OpenTelemetry
+linkTitle: Learn how to instrument Apache Http Server with OpenTelemetry
+date: 2022-05-27
+author: Debajit Das
+---
+
+If you are using Apache Web Server and in dire need of some telemetry tool to monitor your web server, the OpenTelemetry Apache Module is the right candidate for you: it enables tracing of incoming requests to the server and it will capture the response time of many modules (including mod\_proxy) involved in such an incoming request. With that you will get hierarchical time consumption by each module. This article demonstrates the monitoring capabilities of Apache OpenTelemetry Module and quick guide to get started with the module.
+
+## Getting Started with OpenTelemetry Module
+### Building the module
+
+Getting started with the OpenTelemetry module for apache httpd is pretty simple, all you need is a docker engine and git. Download the source code from github and then build the docker image on CentOS7:
+
+```
+git clone https://github.com/open-telemetry/opentelemetry-cpp-contrib
+cd  instrumentation/otel-webserver-module
+docker-compose --profile centos7 build
+```
+
+The above command downloads all dependencies required for building OpenTelemetry module for Apache, builds the module and installs the same on the docker image.
+
+**Note**: The above command might take around 1 hour to complete.
+
+When the build is finished,  run the docker image, by typing the following command:
+```
+docker-compose –profile centos7 up -d
+```
+
+The above command  starts up the centos7 image in a docker container named webserver\_centos7 along with the OpenTelemetry Collector and a Zipkin backend.
+
+Apache OpenTelemetry Module will be configured and installed in the desired location and Apache server will be started with Apache OpenTelemetry Module.
+
+### Viewing spans on the backend
+As mentioned in [docker-compose.yml](https://github.com/open-telemetry/opentelemetry-cpp-contrib/blob/main/instrumentation/otel-webserver-module/docker-compose.yml), webserver\_centos7 listens on port 9004, zipkin on port 9411 and the OpenTelemetry Collector  on port 4317.
+To send  a request to Apache webServer you can either use curl from terminal
+
+```
+curl localhost:9004/
+```  
+Or, you can type ```localhost:9004/``` in any browser.
+
+![Testing](/img/instrument-apache-http-server/testing.png)
+
+Now, traces and spans can be seen on the zipkin backend. To view them, type *localhost:9411* on the browser and click on **Run Query** Button.
+
+![Span-List](/img/instrument-apache-http-server/span-list.png)
+
+This shows a list of queries or endpoints that have been triggered to Apache WebServer.
+e.g */noindex/css*
+
+To see the details  click on any of the **SHOW** buttons.
+
+![Span-Hierarchy](/img/instrument-apache-http-server/span-hierarchy.png)
```

Comment:
Replace `[Span-Hierarchy]` with something speaking [A screenshot from the zipkin UI showing the span hierarchy]